### PR TITLE
DM-31261: Initial push with a project skeleton for the DREAM Common i…

### PR DESCRIPTION
**Pull request by wvreeven**
…nterface library.

### CODE REVIEW

File: doc/.gitignore [hunk 1]
```diff
@@ -1,10 +0,0 @@
-# Doxygen products
```
Comment (tribeiro) [inline doc/.gitignore]: Are you sure you want to remove these?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    "pytest-black",
+    "pytest-mypy",
+]
+dev_requires: List[str] = install_requires + tests_require + ["documenteer[pipelines]"]
+scm_version_template = """# Generated by setuptools_scm
+__all__ = ["__version__"]
+
+__version__ = "{version}"
+"""
+tools_path = pathlib.PurePosixPath(setuptools.__path__[0])
```
Comment (tribeiro) [inline setup.py:41]: I don't think you need the following 3 lines... Consider removing them....

**Review by tribeiro — COMMENTED** [review]
A few inline comments...

File: doc/.gitignore (hunk 1 above)
Comment (wvreeven) [reply doc/.gitignore]: Probably not. I restored the file.

File: setup.py (hunk 2 above)
Comment (wvreeven) [reply setup.py:41]: Done and also two imports that become obsolete because of that.

File: python/lsst/ts/dream/common/abstract_dream.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    async def close_hatch(self) -> None:
+        """Close the hatch."""
+        raise NotImplementedError
+
+    async def stop(self) -> None:
+        """Immediately stop operations and close the hatch."""
+        raise NotImplementedError
+
+    async def ready_for_data(self, ready: bool) -> None:
```
Comment (tribeiro) [inline python/lsst/ts/dream/common/abstract_dream.py]: Small suggestion to rename this method something that would make it more clear its intention. Maybe something like `set_ready_for_data`?

File: python/lsst/ts/dream/common/abstract_dream.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        indicated.
+
+        Parameters
+        ----------
+        ready: `bool`
+            Rubin Observatory is ready to receive data (True) or not (False).
+        """
+        raise NotImplementedError
+
+    async def data_archived(self) -> None:
```
Comment (tribeiro) [inline python/lsst/ts/dream/common/abstract_dream.py]: So, two comments regarding this method. 

First, in general methods should start with a verb that express what the method does (`data` is a noun). I would recommend renaming it to `set_data_archived` or maybe `set_archived_data`.

Next, I am 95.02% certain that we will need to pass in some information to this method so DREAM know which data was archived. If we don't know yet what this data will be, we have two options;

1 - leave it like this but add a comment in the docstring that we have to establish the input.

2 - add a *args, **kwargs  argument and, once we figure out what we want, we make it explicit

File: python/lsst/ts/dream/common/abstract_dream.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["AbstractDream"]
+
+from abc import ABC, abstractmethod
+import typing
```
Comment (tribeiro) [inline python/lsst/ts/dream/common/abstract_dream.py:24]: really picky comment, but consider import typing before abc... 

```
import typing

from abc import ABC, abstractmethod
```

maybe it is just me but I usually prefer to have import organized such that simple imports comes first than "from" and so on...

**Review by tribeiro — APPROVED** [review]
Just left some small picky comments.. Looks good now!

File: python/lsst/ts/dream/common/abstract_dream.py (hunk 5 above)
Comment (wvreeven) [reply python/lsst/ts/dream/common/abstract_dream.py:24]: Done

**Pull request merged**